client/my-sites/plans/current-plan/current-plan-thank-you/thank-you.js

```diff
@@ -4,65 +4,84 @@
 import { connect } from 'react-redux';
 import { getSelectedSiteSlug } from 'state/ui/selectors';
 import { localize } from 'i18n-calypso';
-import React from 'react';
+import React, { Component } from 'react';
 
 /**
  * Internal dependencies
  */
+import { isDesktop } from 'lib/viewport';
 import { preventWidows } from 'lib/formatting';
+import { requestGuidedTour } from 'state/ui/guided-tours/actions';
 import Button from 'components/button';
 
 import './style.scss';
 
-export function ThankYouCard( {
-	children,
-	illustration,
-	showCalypsoIntro,
-	showContinueButton,
-	showHideMessage,
-	siteSlug,
-	title,
-	translate,
-} ) {
-	return (
-		<div className="current-plan-thank-you">
-			{ illustration && (
-				<img
-					alt=""
-					aria-hidden="true"
-					className="current-plan-thank-you__illustration"
-					src={ illustration }
-				/>
-			) }
-			{ title && <h1 className="current-plan-thank-you__title">{ title }</h1> }
-			{ children }
-			{ showCalypsoIntro && (
-				<p>
-					{ preventWidows(
-						translate(
-							'This is your new WordPress.com dashboard. You can manage your site ' +
-								'here, or return to your self-hosted WordPress dashboard using the ' +
-								'link at the bottom of your checklist.'
-						)
-					) }
-				</p>
-			) }
-			{ showContinueButton && (
-				<Button primary href={ `/plans/my-plan/${ siteSlug }` }>
-					{ translate( 'Continue' ) }
-				</Button>
-			) }
-			{ showHideMessage && (
-				<p>
-					<a href={ `/plans/my-plan/${ siteSlug }` }>{ translate( 'Hide message' ) }</a>
-				</p>
-			) }
-		</div>
-	);
+export class ThankYouCard extends Component {
```

Comment:
@sirreal I switched this back to class Component in order to get `requestGuidedTour` functioning. Lemme know if there's another way. :-)

Reply:
There are other ways, each with their own tradeoffs. This approach is perfectly fine, I'm not too concerned with whether we have function or class components generally.